core/src/main/java/com/linecorp/armeria/server/ServerHttp2ObjectEncoder.java

```diff
@@ -143,7 +145,7 @@ public void maybeResetStream(int streamId, Http2Error http2Error) {
         if (stream == null) {
             return;
         }
-        if (stream.state().localSideOpen() || stream.state().remoteSideOpen()) {
+        if ((stream.state().localSideOpen() && !isEndStreamSent(stream)) || stream.state().remoteSideOpen()) {
```

Comment:
Question) In what situation can `localSideOpen()` be true when this method is called?

Reply:
From my understanding, this method is called when `resWriteFuture` completes. It means the `HttpResponse` was fully sent to the Netty channel.
So I was wondering if we can remove `(stream.state().localSideOpen() && !isEndStreamSent(stream))` condition and check only `remoteSideOpen()`.
```suggestion
        if (stream.state().remoteSideOpen()) {
```

Reply:
Good point, I agree the remote side should more or less always be in sync, so we may only need to send the RST frame when local side is open only.
Updated.